content/howto/extensibility/how-to-create-a-native-pluggable-widget.md

```diff
@@ -0,0 +1,1344 @@
+![Pluggable Widget Generator input example](attachments/how-to-create-a-native-pluggable-widget/pluggable-widget-generator-input-example.png)
+
+### 3.2 Creating a Mendix test project
+
+In order to test our group box widget we want to create a Mendix application that uses this widget in various ways.
+
+1. Open Mendix Studio Pro v8.2.2 to create a new test project by selecting **File > New Project** from the top menu bar. Select the starter app **Blank**, which is the last option on the last page. Then click the **Use this starting point** button, and fill in the following details in the **App Settings** dialog box:
+
+   - App name: **GroupBoxTestProject**
+   - Enable online services: **No**
+
+   After that, click the **Create app** button.
+
+   Optionally you may remove all unused custom widgets to optimize the debugging process. Select **Project > Show Project Directory in Explorer** from the Mendix Studio Pro menu and open the **widgets** folder. Then, delete all the files in this folder.
+
+   After that, close the project by selecting **File > Close Project** from the top menu bar.
+
+2. Move the contents of the Mendix test project folder into **test/MxTestProject** which is located inside your group box widget project folder.
+
+3. Open the moved Mendix test project by selecting **File > Open Project...** from the top menu bar. In the **Open App** dialog box, select **Locally on disk** and browse to the **GroupBoxTestProject.mpr** file inside your group box widget project folder.
+
+4. Create a test case for the group box widget:
+
+   1. Open **Home_Native** page.
+   2. Copy "Home" container and paste it underneath that container
+   3. Delete the content.
+   4. Search in toolbox for the **Group Box** widget.
+   5. Drag the widget into the new container.
+   6. Double click the widget and fill in your name as default value.
+
+5. Setting up the test project is done. Run the test project locally and use the makeitnative app to view the sample widget code in action. If you followed the steps carefully you should see yellow text in your app saying "Hello {your name}".
+
+### 3.3 Building the group box widget
+
+Now that we have seen our project code running in native, it is time to transform our sample widget into an actual group box widget.
+
+Open up your group box widget project folder in your IDE of choice. From now on, all file references will be relative to this project folder.
+
+#### 3.3.1 Creating static UI
+
+First we will change the display component. A display component represents the widget and doesn't interact with Mendix specific APIs and, therefore, can be used in any React application. The current display component is located in **src/components** and has the filename **HelloWorldSample.tsx**. Open this file and you can see a typical React Native component with a render method. This method returns the Graphical User Inteface (GUI) of the component.
+
```

Comment:
Show intermediate steps to confirm you are still up to date with the sample
- show generated widget works.
- create structure, show success
- Add style + image
- Add os style + images

etc..